src/web3/KeychainAccountManager.ts

```diff
@@ -194,43 +188,16 @@ export class KeychainSigner implements Signer {
     await storePrivateKey(privateKey, this.account, passphrase)
   }
 
-  async signTransaction(
-    addToV: number,
-    encodedTx: RLPEncodedTx
-  ): Promise<{ v: number; r: Buffer; s: Buffer }> {
-    Logger.info(`${TAG}@signTransaction`, `Signing transaction:`, encodedTx.transaction)
-    const { gasPrice } = encodedTx.transaction
-    const gasPriceBN = new BigNumber((gasPrice || 0).toString())
-    if (gasPriceBN.isNaN() || gasPriceBN.isLessThanOrEqualTo(0)) {
-      // Make sure we don't sign and send transactions with 0 gas price
-      // This resulted in those TXs being stuck in the txpool for nodes running geth < v1.5.0
-      throw new Error(`Preventing sign tx with 'gasPrice' set to '${gasPrice}'`)
-    }
-
-    return this.localSigner.signTransaction(addToV, encodedTx)
-  }
-
-  async signPersonalMessage(data: string): Promise<{ v: number; r: Buffer; s: Buffer }> {
-    Logger.info(`${TAG}@signPersonalMessage`, `Signing ${data}`)
-    return this.localSigner.signPersonalMessage(data)
-  }
-
-  async signTypedData(typedData: EIP712TypedData): Promise<{ v: number; r: Buffer; s: Buffer }> {
-    Logger.info(`${TAG}@signTypedData`, `Signing typed DATA:`, { address: this.account, typedData })
-    return this.localSigner.signTypedData(typedData)
-  }
-
-  getNativeKey() {
-    return this.account.address
-  }
-
   async unlock(passphrase: string, duration: number): Promise<boolean> {
```

Comment:
I considered doing per-wallet locks, but the disadvantage of that is that for a considerable amount of time we will have some functionality in the app on an Ethers wallet and some on a contractkit wallet. I wanted them to share the same lock so that the user experience isn't any different than it is now.

Reply:
Are you saying that we might want to have per-chain locking/unlocking down the road? I think that could be easily implemented here later if we want.

Reply:
seems simpler to unlock everything and get more granular if we discover a need for it